src/middleware/validateUser.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+import { expectedNewUser, testPassword } from '../utils/validations';
+import sendResponse from '../utils/response';
+
+export const isNewUserValid = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { error } = expectedNewUser.validate(req.body);
+  if (error) {
+    return sendResponse<null>(res, 400, null, error.message);
+  }
+
+  const passwordError = testPassword(req.body.password);
+  if (passwordError) {
+    return sendResponse<null>(res, 400, null, passwordError);
+  }
+
+  next();
+};
```

Comment:
also you and @Ruthuwamahoro you have to harmonize this help function where it will be used by everyone if you need help from this I can help you tomorrow make it resuable

Reply:
I have refactored this function